prov/gni/src/gnix_nic.c

```diff
@@ -1168,6 +1172,12 @@ int gnix_nic_alloc(struct gnix_fid_domain *domain,
 					  (size_t)nic->mem_per_mbox,
 					  domain->params.mbox_num_per_slab,
 					  &nic->mbox_hndl);
+
+		/*
+		 * TODO: Create a buddy allocator on the nic for iov
+		 * operations
+		 */
```

Comment:
I do not think that we should assume we'll need the buddy allocator here before we know what the use case is and if it's actually a problem to use regular ol' malloc.  Please open an issue to investigate whether we'll need it once people start using sendv/recvv.